src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageReadmeControl.xaml.cs

```diff
@@ -87,7 +85,14 @@ private void PackageReadmeControl_Unloaded(object sender, RoutedEventArgs e)
 
         private void PackageReadmeControl_Loaded(object sender, RoutedEventArgs e)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(UpdateMarkdownAsync);
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                var componentModel = await AsyncServiceProvider.GlobalProvider.GetComponentModelAsync();
+                var markdownPreviewSingleton = componentModel.GetService<MarkdownPreviewSingleton>();
+                _markdownPreview = markdownPreviewSingleton.GetInstance();
```

Comment:
The singleton could be null, right?

Reply:
We don't consistently check for null for objects retrieved using `componentModel.GetService`. My gut says it shouldn't happen though since MarkdownPreviewSingleton is set up using MEF. I can make a change to hade the tab if markdownPreviewSingleton is null if you think it's necessary.

Reply:
I haven't seen it happen in a while, but until a year or two ago, each year I'd see at least 3 or 4 customers report that NuGet in VS was broken. Turned out that the MEF cache was broken and repairing VS fixed it, but since NuGet was the first component in VS that needed MEF, so NuGet was the first thing the customer noticed was broken.

So, on one hand, PM UI probably won't even load if the MEF cache is completely busted, so by the time it gets to this code, the MEF cache will probably successfully resolve this component.

On the other hand, there's been enough MEF problems over the years that I feel like it's only a matter of time until it gets a null (even if it's only in the tiniest percentage of customers), then line 92 causes a null ref exception.  Since this code is in a WPF callback, we know it's running on the UI thread, and an unhandled exception on the UI thread is going to cause a crash to desktop.